client/testserver/src/main/scala/sttp/tapir/client/tests/HttpServer.scala

```diff
@@ -50,23 +49,25 @@ class HttpServer(port: Port) {
       if (f == "papaya") {
         Accepted("29")
       } else {
-        Ok(s"fruit: $f${amount.map(" " + _).getOrElse("")}", Header("X-Role", f.length.toString))
+        Ok(s"fruit: $f${amount.map(" " + _).getOrElse("")}", Header.Raw(CIString("X-Role"), f.length.toString))
       }
     case GET -> Root / "fruit" / f                                         => Ok(s"$f")
     case GET -> Root / "fruit" / f / "amount" / amount :? colorOptParam(c) => Ok(s"$f $amount $c")
     case _ @GET -> Root / "api" / "unit"                                   => Ok("{}")
     case r @ GET -> Root / "api" / "echo" / "params"                       => Ok(r.uri.query.params.toSeq.sortBy(_._1).map(p => s"${p._1}=${p._2}").mkString("&"))
     case r @ GET -> Root / "api" / "echo" / "headers" =>
-      val headers = r.headers.toList.map(h => Header(h.name.value, h.value.reverse))
-      val filteredHeaders = r.headers.find(_.name.value.equalsIgnoreCase("Cookie")) match {
-        case Some(c) => headers.filter(_.name.value.equalsIgnoreCase("Cookie")) :+ Header("Set-Cookie", c.value.reverse)
+      val headers = r.headers.headers.map(h => Header.Raw(CIString(h.name.toString), h.value.reverse))
+      val filteredHeaders: List[Header.Raw] = r.headers.headers.find(_.name.toString.equalsIgnoreCase("Cookie")) match {
+        case Some(c) => headers.filter(_.name.toString.equalsIgnoreCase("Cookie")) :+ Header.Raw(CIString("Set-Cookie"), c.value.reverse)
         case None    => headers
       }
-      Ok(headers = filteredHeaders: _*)
+      okOnlyHeaders(filteredHeaders.map(x => x: Header.ToRaw))
```

Comment:
@catostrophe if you'd have a better idea how to handle this case I'd be happy to take hints :) Here I've get a list of raw headers and I want to send them back (and only the headers). I couldn't get it to work without the (otherwise useless) map with a cast (triggering the implicit conversion), but it's probably because of friday evening not because it's necessary ;)

Reply:
```scala
    case r @ GET -> Root / "api" / "echo" / "headers" =>
      val headers = r.headers.headers.map(h => h.copy(value = h.value.reverse))
      val filteredHeaders: Header.ToRaw = r.headers.headers.find(_.name == CIString("Cookie")) match {
        case Some(c) => headers.filter(_.name == CIString("Cookie")) :+ Header.Raw(CIString("Set-Cookie"), c.value.reverse)
        case None    => headers
      }
      okOnlyHeaders(List(filteredHeaders))
    case r @ GET -> Root / "api" / "echo" / "param-to-header" =>
      okOnlyHeaders(r.uri.multiParams.getOrElse("qq", Nil).reverse.map("hh" -> _: Header.ToRaw))
    case r @ GET -> Root / "api" / "echo" / "param-to-upper-header" =>
      okOnlyHeaders(r.uri.multiParams.map { case (k, v) =>
        k -> v.headOption.getOrElse("?"): Header.ToRaw
      }.toSeq)
```

Reply:
I would write it this way

Reply:
Thanks, much better :)